src/wwwroot/services/auth.js

```diff
@@ -33,30 +33,58 @@
       getLimitsByAccount: getLimitsByAccount,
       getFreeTrialNotificationFromStorage: getFreeTrialNotificationFromStorage,
       addFreeTrialNotificationToStorage: addFreeTrialNotificationToStorage,
-      changeEmail: changeEmail
+      changeEmail: changeEmail,
+      decodeToken: decodeToken
     };
 
     var decodedToken = null;
+    var permissons = null;
     var encodedToken = null;
     var temporarilyAuthed = false;
 
     encodedToken = $window.localStorage.getItem('jwtToken');
     if (encodedToken) {
-      decodedToken = jwtHelper.decodeToken(encodedToken);
+      decodeToken();
       // verify expiration
     }
     return authService;
 
     // Save the token in the local storage (globally) or in a temporal variable (local)
     function saveToken(token, isTemporal) {
-      decodedToken = jwtHelper.decodeToken(token);
       encodedToken = token;
+      decodeToken();
       temporarilyAuthed = !!isTemporal;
       if (!temporarilyAuthed) {
         $window.localStorage.setItem('jwtToken', token);
       }
     }
 
+    function decodeToken() {
+      decodedToken = jwtHelper.decodeToken(encodedToken);
+      if (!decodedToken.profile) {
+        return permissons = null;
```

Comment:
I think that it is not a good idea to hay a function that returns value and also generates side effects. 

What do you think about considering _permissons_ as part of decoded token?

```js
function decodeToken() {
  decodedToken = jwtHelper.decodeToken(encodedToken);
  decodedToken.permissons = expandPermissons(decodedToken.profile);
}

function expandPermissons(profile) {
  // If switch works well with null, this condition could be removed
  if (!profile) {
    return null;
  }
  switch (profile) {
    case "reports": return {
      acceptedUrlsPattern: /^#?\/reports\/?(\?.*)?$|^#?\/reports\/downloads\/?(\?.*)?$/,
      defaultUrl: "/reports"
    };
    case "templates": return {
      acceptedUrlsPattern: /^#?\/templates\/?(\?.*)?$/,
      defaultUrl: "/templates"
    };
    case "settings": return {
      acceptedUrlsPattern: /^#?\/settings\//,
      defaultUrl: "/settings/connection-settings"
    };
    default: return null;
}
```